### PR TITLE
[Tailcall] Dial back single -v output, and enable under MONO_LOG_LEVEL=debug MONO_LOG_MASK=tailcall.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -3581,6 +3581,8 @@ mini_parse_debug_option (const char *option)
 	// It is asserted.
 	else if (!strcmp (option, "test-tailcall-require"))
 		mini_debug_options.test_tailcall_require = TRUE;
+	else if (!strcmp (option, "tailcall"))
+		mini_debug_options.tailcall = TRUE;
 	else
```
Comment (vargaz) [inline mono/mini/mini-runtime.c]: This might be better named 'log-tailcall' or something.

Comment (jaykrell) [reply mono/mini/mini-runtime.c]: I thought of that and am ok with that but I figured printing is the most basic of debugging feature. Perhaps that is my bias.

File: mono/mini/method-to-ir.c [hunk 2]
```diff
@@ -2176,7 +2175,7 @@ test_tailcall (MonoCompile *cfg, MonoBoolean tailcall)
 	// Do not change "tailcalllog" here without changing other places, e.g. tests that search for it.
 	//
 	g_assertf (tailcall || !mini_get_debug_options ()->test_tailcall_require, "tailcalllog fail from %s", cfg->method->name);
-	if (cfg->verbose_level)
+	if (cfg->verbose_level > 1 || mini_get_debug_options ()->tailcall)
```
Comment (luhenry) [inline mono/mini/method-to-ir.c]: We should use `mono_trace` instead, it's exactly done for these kind of use cases.

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: I see printf and g_print a lot and was unaware. Let me look more then.

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: MONO_LOG_LEVEL=debug MONO_LOG_MASK=tailcall and I fill in flag_mask_map? I'll try that.
ok to also support -v -v, or we should stop adding anything under verbose_level?

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: oh, no, can't really have -v -v also, that'd be messy. just mono_trace.

Comment (jaykrell) [reply mono/mini/method-to-ir.c]: notice method-to-ir.c never uses mono_trace..thus my mistake

**Pull request merged**